GoogleForCreators/web-stories-wp
# Media: Initial implementation for captions hotlinking

## Context



## Summary

Add a basic implement of to allow users to hotlink caption files for audio / video element. 

## Relevant Technical Choices

Created a reusable component that can be used for other parts of the code that might need hotlinking support. 

## To-do



## User-facing changes

https://user-images.githubusercontent.com/237508/161045350-f28b6eb5-5e2b-47f9-a58e-aa138d55893a.mov

## Testing Instructions






This PR can be tested by following these steps:

1. Enable caption hotlinking experiment. 
2. Create a story. 
3. Insert a video. 
4. Go to caption settings. 
5. Click Link to file. 
6. See dialog. 
7. Try input invalid urls. 
8. Click insert. 
9. See error. 
10. Insert valid url. example https://assets.computingcorner.co.uk/uploads/2022/03/test.vtt
11. Click insert. 
12. See captions functionality work correctly. 


## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Fixes #11081
